x/axelarnet/message_handler.go

```
@@ -226,7 +229,7 @@ func validateMessage(ctx sdk.Context, ibcK keeper.IBCKeeper, n types.Nexus, b ty
 			return fmt.Errorf("asset %s is not registered on chain %s", token.GetDenom(), srcChain.Name)
 		}
 
-		if !n.IsAssetRegistered(ctx, destChain, token.GetDenom()) {
+		if found && !n.IsAssetRegistered(ctx, destChain, token.GetDenom()) {
```

tokens can't be sent to amplifier, so this should always fail if it's not found